tests/infrared/metrics-collectd-qdr.yaml.template

```diff
@@ -4,28 +4,45 @@ tripleo_heat_templates:
 
 custom_templates:
     parameter_defaults:
-        CollectdAmqpInterval: 1
-        CollectdDefaultPollingInterval: 1
-        CollectdExtraPlugins:
+        CollectdAmqpInterval: 5
+        CollectdDefaultPollingInterval: 5
+        CollectdDefaultPlugins:
```

Comment:
Could you please change this back to CollectdExtraPlugins?

CollectdDefaultplugins will just replace the defaults, ExtraPlugins will append the list of defaults.
Not saying this will cause an issue here, but it lead to an escalation before, as healthchecks rely on the unixsock plugin, which was removed in DefaultPlugins.

Reply:
No, I disagree with your thinking here.

This file in specific is the driver for our integration tests and should produce a specific known set of plugins, independent of the upstream list so that tests are on par. No customer is ever going to file an escalation because of this file.

Part of the purpose of this PR is to help @ryan-mccabe decide on exactly which of these should be in the upstream to begin with. That is independent work, but it should be well informed by the discussion here.

Reply:
And now this PR makes way more sense to me. OK so the changes here are basically going to be a drop in setup for the upstream file when specifically enabling the "SAF environment" (the file is `environment/enable-saf.yml`).

In this case yes I agree with @csibbitt in that we want to override the generic defaults for collectd because in our case, we want to clearly define exactly what we're enabling and how. 

What would cause an escalation is that someone changes the default plugins on us (which we would otherwise rely on) and enables a plugin that overwhelms the SmartGateway. In that case we'd get an escalation on cloud update if that were to happen.

The environment is being setup specifically for data collection with SAF, so yea, let's make it implicit here so we don't have to debug multiple layers.

Reply:
I retract my prior thumbs up and have moved it to another location.

Reply:
I still don't think we're all on the same page here. I am not trying to defend the use of `CollectdDefaultplugins` in enable-saf.yaml - I think that's up for debate and Leif and Matthias seem to be on opposite side of that debate, currently. I'm undecided. This is the wrong code review for that debate.

I'm defending it's use HERE, in this file, because this is the definition for a test harness and I was specifically disabling some of the upstream plugins in order to achieve the configuration that I want to propose. THIS is the set of plugins I tested, observed, reported on, and am proposing. Not this set plus the default set.

Now there's also a reasonable argument that this file should eventually go away and we should use enable-saf.yaml for our testing (deterministic or not), because that's what the customer will get. That seems right to me, but that file doesn't exist yet, and the change to add it was not even open when I wrote this.

I hope that clarifies what we're talking about.

Reply:
Yes that helps clarify it further. In that case, then I think this is perfectly fine to merge and you should do so if there are no further changes to make.

Around the debate as to whether this should match upstream (`enable-saf.yml`) we should probably just setup a small meeting to determine our approach there. I suspect the defaults listed here are going to directly affect Ryan's upstream PR. We should make sure the knowledge from here flows up into that, regardless as to whether we use the `Default` or `Extra` approach. Agreed we can figure that out separately.

I'm going to approve this since I have no further feedback.

Reply:
Okay, following that discussion, use of `CollectdDefaultPlugins` makes sense.

However, I wanted to highlight the general bad practice of using CollectdDefaultPlugins in tripleo yaml files for configuration, unless you exactly know what you are doing, especially when using the product. I'm speaking from experience, I already debugged this during a customer escalation.